tox.ini

```diff
@@ -1,7 +1,8 @@
 [tox]
-minversion = 2.1
+minversion = 3.23.0
 envlist = py37, py38, py39, py310, lint-incr
-skipsdist = True
+isolated_build = true
+requires = setuptools-rust>=1.1.0
```

Comment:
If we need `setuptools-rust>=1.1.0` we should probably encode that in `requirements-dev.txt` and `pyproject.toml` as well.

Reply:
Oh, I probably should come in and clean up the tox.ini here. Most of this isn't actually needed, it was my failed attempt to fix the docs job which was failing on not finding setuptools-rust. But it wasn't a tox issue, instead I missed a place setup.py was being called explicitly for no reason in the azure job def.

Reply:
Removed in: https://github.com/Qiskit/qiskit-terra/pull/7658/commits/4c4c9d4e0c0e98637965994d0bbbb28a552e7c39